File: Java Spring Boot Project/CompanyDirectory/src/main/java/com/mindthecode/CompanyDirectory/models/entities/Position.java
package com.mindthecode.CompanyDirectory.models.entities;

import com.mindthecode.CompanyDirectory.common.Enums;

import javax.persistence.*;

@Entity
@Table(name = "Positions")
public class Position {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String name;

    @ManyToOne
    private Unit unit;

    public Position() {
    }

    public Position(String name) {
        this.name = name;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Unit getUnit() {
        return unit;
    }

    public void setUnit(Unit unit) {
        this.unit = unit;
    }


    public class AvailablePositions {
        public class JuniorAnalyst {
        }
    }
}

File: Java Spring Boot Project/company_directory_app/lib/models/entities/position.dart
import 'package:company_directory_app/models/entities/unit.dart';

class Position {
  final int id;
  final String name;
  final Unit unit;

  Position({this.id, this.name, this.unit});

  factory Position.fromJson(Map<String, dynamic> json) {
    return Position(id: json['id'], name: json['name'], unit: Unit.fromJson(json['unit']));
  }
}
